### DISCUSSION TITLE
Traits from Postgres Extensions

### DISCUSSION THREAD

**Discussion by dmgolembiowski** [geldata/gel#2037] [Ideas]
I'm wondering if traits can be implemented on top of postgres plugins like the one demonstrated [here](https://github.com/LI3DS/pg-li3ds/blob/master/extension/li3ds--1.0.0.sql).
This stems from the source's curious implementation that leverages `$CODE$` blocks to inject native language bindings into the function body.

```sql
create or replace function transform(box4d libox4d, transfo integer, ttime text)
returns libox4d as
$CODE$
    import pg_li3ds
    return pg_li3ds.transform_box4d_one(box4d, transfo, ttime)
$CODE$ language plpython2u;
```

**Comment by elprans** [comment]
We have plans for an extension API to allow, among other things, the wrapping of Postgres extensions.  This isn't a priority for 1.0, though.
